packages/consumption/src/modules/requests/incoming/IncomingRequestsController.ts

```diff
@@ -183,6 +183,23 @@ export class IncomingRequestsController extends ConsumptionBaseController {
 
         const relationship = await this.relationshipResolver.getRelationshipToIdentity(request.peer);
         // It is safe to decide an incoming Request when no Relationship is found as this is the case when the Request origins from onNewRelationship of the RelationshipTemplateContent
+
+        if (relationship?.peerDeletionInfo?.deletionStatus === PeerDeletionStatus.Deleted) {
+            return ValidationResult.error(
+                ConsumptionCoreErrors.requests.peerIsDeleted(
+                    `You cannot decide a Request from '${request.peer.toString()}' since the peer is in status '${relationship.peerDeletionInfo.deletionStatus}'.`
```

Comment:
Since you distinguish both cases here, there is no need to pass the deletion status as variable. You can simple write "... since the peer is deleted."